//python -m grpc_tools.protoc --proto_path=./proto --python_out=./pb --grpc_python_out=./pb product.proto

syntax = "proto3";
package product;


message Product {
    string sku = 1;
    string name = 2;

    message Images {
        string small = 1;
        string large = 2;
    }

    Images images = 3;
}

message GetProductsBySkusRequest {
    repeated string skus = 1;
}

message GetProductsRequest {
    string term = 1;
}

message Products {
    repeated Product product = 1;
}

service ProductService {
    rpc GetProducts (GetProductsRequest) returns (Products);
    rpc GetProductsBySkus (GetProductsBySkusRequest) returns (Products);
}

